File: browserControl/js/auto.js
/**
* Commands to use with automomy module
* @module control/auto
*/
import {RemoteEvent, RemoteVarEvent} from "./remoteEvent.js";

/**Triggers to start automation
 * @prop {bool} [hasGravel] - is the robot currently carring gravel
 */
const startAuto = new RemoteEvent("startAuto")

/**Triggers to stop automation
 */
const stopAuto = new RemoteEvent("stopAuto")

/**Current status of the automaton module
 * @prop TODO
 */
const autoStatus = new RemoteVarEvent("autoStatus", {})

export {startAuto, stopAuto, autoStatus};

File: robot/auto.py
'''Manages autonomus operation
Uses location to handle finding pit location and moveing robot
Needs to send telemetry info to clients
'''
import location
import remoteEvent

'''Triggers to start automation

    Property:
        hasGravel (bool, optional): is the robot currently carring gravel
'''
startAuto = remoteEvent.RemoteEvent("startAuto")

'''Triggers to stop automation
'''
stopAuto = remoteEvent.RemoteEvent("stopAuto")

'''Current status of the automaton module

    Property:
        TODO
'''
autoStatus = remoteEvent.RemoteVarEvent("autoStatus", {})